cmd/foghorn/main.go

```diff
@@ -56,28 +52,23 @@ func main() {
 		logrus.WithError(err).Fatal("Could not create kubeconfig")
 	}
 
-	lhClient, err := clientset.NewForConfig(cfg)
-	if err != nil {
-		logrus.WithError(err).Fatal("Could not create Lighthouse API client")
-	}
-	kubeClient, err := kubernetes.NewForConfig(cfg)
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{Scheme: scheme, Namespace: o.namespace})
 	if err != nil {
-		logrus.WithError(err).Fatal("Could not create Kubernetes API client")
+		logrus.WithError(err).Fatal("Unable to start manager")
 	}
-	lhInformerFactory := lhinformers.NewSharedInformerFactoryWithOptions(lhClient, time.Minute*30, lhinformers.WithNamespace(o.namespace))
-
-	controller, err := foghorn.NewController(kubeClient,
-		lhClient,
-		lhInformerFactory.Lighthouse().V1alpha1().LighthouseJobs(),
-		o.namespace,
-		nil)
 
+	reconciler, err := foghorn.NewLighthouseJobReconciler(mgr.GetClient(), mgr.GetScheme(), o.namespace, nil, nil, nil, nil)
 	if err != nil {
-		logrus.WithError(err).Fatal("Error creating controller")
+		logrus.WithError(err).Fatal("Unable to instantiate reconciler")
+	}
+	if err = reconciler.SetupWithManager(mgr); err != nil {
+		logrus.WithError(err).Fatal("Unable to create controller")
 	}
-	lhInformerFactory.Start(stopCh)
 
-	if err = controller.Run(2, stopCh); err != nil {
-		logrus.WithError(err).Fatal("Error running controller")
+	defer reconciler.ConfigMapWatcher.Stop()
+
+	defer interrupts.WaitForGracefulShutdown()
```

Comment:
I don't think this does anything anymore in this case. This was mainly working for the HTTP servers started/stopped via the functions in the `interrupts` package. But since we are not using any of this anymore, there is no need to call this. 

It seems controller-runtime has it own helper/api to handle the shutdown singnals:

```
mgr.Start(ctrl.SetupSignalHandler())
```

Reply:
Yup, should have deleted this.